src/rrt/Tree.hpp

```diff
@@ -408,7 +478,10 @@ class Tree {
         reset(true);
 
         //  create root node from provided start state
-        _nodes.push_back(Node<T>(startState, nullptr));
+        Node<T>* root =
+            new Node<T>(startState, nullptr, _dimensions, _TToArray);
```

Comment:
Um _nodes is a deque. of Nodes not of pointers. This is actually creating a Node on the stack, copying it into the array and leaving it on the stack.

Just change it to Node root(startState, nullptr, _dimensions, _TToArray);
and std::move it it in. Even better just do 
_nodes.emplace_backe(startState, nullptr, _dimensions, _TToArray);
Thats the fastest and doesn't it only gets inserted once.